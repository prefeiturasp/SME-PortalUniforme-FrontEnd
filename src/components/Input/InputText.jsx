import PropTypes from "prop-types";
import React from "react";
import { InputErroMensagem } from "./InputErroMensagem";
import { HelpText } from "components/HelpText";
import "./style.scss";

export class InputText extends React.Component {
  render() {
    const {
      className,
      disabled,
      esconderAsterisco,
      helpText,
      input,
      label,
      labelClassName,
      meta,
      min,
      name,
      placeholder,
      required,
      type,
      maxLength,
      customChange,
      autoFocus
    } = this.props;
    return (
      <div className="input">
        {label && [
          required && !esconderAsterisco && (
            <span key={1} className="required-asterisk">
              *
            </span>
          ),
          <label
            key={2}
            htmlFor={name}
            className={`col-form-label ${labelClassName}`}
          >
            {label}
          </label>
        ]}
        <input
          {...input}
          autoFocus={autoFocus}
          className={`form-control ${className} ${meta.touched &&
            meta.error &&
            "invalid-field"}`}
          disabled={disabled}
          min={min}
          name={name}
          placeholder={placeholder}
          required={required}
          type={type}
          maxLength={maxLength}
          {...(() => {
            if (customChange) {
              return {
                onChange: e => {
                  input.onChange(customChange(e.target.value));
                }
              };
            } else {
              return {};
            }
          })()}
        />
        <HelpText helpText={helpText} />
        <InputErroMensagem meta={meta} />
      </div>
    );
  }
}

InputText.propTypes = {
  className: PropTypes.string,
  disabled: PropTypes.bool,
  esconderAsterisco: PropTypes.bool,
  helpText: PropTypes.string,
  input: PropTypes.object,
  label: PropTypes.string,
  labelClassName: PropTypes.string,
  meta: PropTypes.object,
  name: PropTypes.string,
  placeholder: PropTypes.string,
  required: PropTypes.bool,
  type: PropTypes.string
};

InputText.defaultProps = {
  className: "",
  disabled: false,
  esconderAsterisco: false,
  helpText: "",
  input: {},
  label: "",
  labelClassName: "",
  meta: {},
  name: "",
  placeholder: "",
  required: false,
  type: "text"
};
