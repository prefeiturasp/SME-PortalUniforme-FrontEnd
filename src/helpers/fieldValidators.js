export const required = value =>
  value !== undefined ? undefined : "Campo obrigatório";

export const email = value =>
  value && !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(value)
    ? "Email inválido"
    : undefined;

export const prefeituraEmail = value =>
  value && /.+@\prefeitura.sp.gov.br/.test(value)
    ? undefined
    : "Somente emails da prefeitura de São Paulo";

export const alphaNumeric = value =>
  value && /[^a-zA-Z0-9 ]/i.test(value)
    ? "Only alphanumeric characters"
    : undefined;

export const numericInteger = value =>
  value && /[^0-9 ]/i.test(value) ? "Somente números" : undefined;

export const phoneNumber = value =>
  value && !/^(0|[1-9][0-9]{9})$/i.test(value)
    ? "Invalid phone number, must be 10 digits"
    : undefined;
