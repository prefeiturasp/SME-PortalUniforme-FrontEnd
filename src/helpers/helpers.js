const STATUS_DICT = {
  INSCRITO: "Inscrição finalizada, em processo de credenciamento",
  CREDENCIADO: "Credenciado",
  EM_PROCESSO: "Em processo de inscrição, finalize o seu cadastro",
  PENDENTE: "Pendente credenciamento",
  ALTERADO: "Pendente validação de alteração",
};

export const getStatus = (status) => {
  return STATUS_DICT[status] || status;
};

export const getCNPJ = () => {
  return localStorage.getItem("cnpj");
};

export const getRazaoSocial = () => {
  return localStorage.getItem("razao_social");
};

export const getKey = (obj) => {
  return Object.keys(obj)[0];
};

export const getError = (obj) => {
  let result = "Erro";
  if (!obj[getKey(obj)]) {
    return "Erro";
  } else if (
    (obj[getKey(obj)][0] !== undefined &&
      typeof obj[getKey(obj)][0] !== "string") ||
    typeof obj[getKey(obj)] !== "string"
  ) {
    result = getError(obj[getKey(obj)]);
  } else {
    if (typeof obj[getKey(obj)] === "string") return obj[getKey(obj)];
    else return obj[getKey(obj)][0];
  }
  return result;
};
