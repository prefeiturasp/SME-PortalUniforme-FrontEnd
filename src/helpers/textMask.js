import { createTextMask } from "redux-form-input-masks";

export const fieldCPF_CNPJ = value => {
  if (value.length <= 11) {
    return value
      .replace(/\D/g, "")
      .replace(/(\d{3})(\d{3})(\d{3})(\d{2})/g, "$1.$2.$3-$4");
  } else {
    return value
      .replace(/\D/g, "")
      .replace(/(\d{2})(\d{3})(\d{3})(\d{4})(\d{2})/g, "$1.$2.$3/$4-$5");
  }
};

export const fieldCep = createTextMask({
  pattern: "99999-999",
  allowEmpty: false,
  guide: true,
  stripMask: false
});

export const fieldTel = createTextMask({
  pattern: "(99) 99999-9999",
  allowEmpty: false,
  guide: true,
  stripMask: false
});

export const fieldCel = createTextMask({
  pattern: "(99) 99999-9999",
  allowEmpty: false,
  guide: true,
  stripMask: false
});
