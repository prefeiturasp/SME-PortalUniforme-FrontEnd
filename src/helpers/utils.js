export async function asyncForEach(array, callback) {
  for (let index = 0; index < array.length; index++) {
    await callback(array[index], index, array);
  }
}

export async function readerFile(file) {
  let result_file = await new Promise(resolve => {
    const reader = new FileReader();
    reader.onload = () => {
      const re = /(?:\.([^.]+))?$/;
      const base64 = reader.result.split("base64,")[1];
      return resolve({
        planta: `data:${file.type}/${re.exec(file.name)[1]};base64,${base64}`
      });
    };
    reader.readAsDataURL(file);
  });
  return result_file;
}

export const getKey = obj => {
  return Object.keys(obj)[0];
};

export const getError = obj => {
  let result = "Erro ao cadastrar Imóvel";
  if (!obj[getKey(obj)]) {
    return "Erro ao cadastrar Imóvel";
  } else if (
    (obj[getKey(obj)][0] !== undefined &&
      typeof obj[getKey(obj)][0] !== "string") ||
    typeof obj[getKey(obj)] !== "string"
  ) {
    result = getError(obj[getKey(obj)]);
  } else {
    if (typeof obj[getKey(obj)] === "string") return obj[getKey(obj)];
    else return obj[getKey(obj)][0];
  }
  return result;
};

export const validarCPF = cpf => {
  cpf = cpf.replace(/[^\d]+/g, "");
  if (cpf === "") return false;
  // Elimina CPFs invalidos conhecidos
  if (
    cpf.length !== 11 ||
    cpf === "00000000000" ||
    cpf === "11111111111" ||
    cpf === "22222222222" ||
    cpf === "33333333333" ||
    cpf === "44444444444" ||
    cpf === "55555555555" ||
    cpf === "66666666666" ||
    cpf === "77777777777" ||
    cpf === "88888888888" ||
    cpf === "99999999999"
  )
    return false;
  // Valida 1o digito
  let add = 0;
  let i, rev;
  for (i = 0; i < 9; i++) add += parseInt(cpf.charAt(i)) * (10 - i);
  rev = 11 - (add % 11);
  if (rev === 10 || rev === 11) rev = 0;
  if (rev !== parseInt(cpf.charAt(9))) return false;
  // Valida 2o digito
  add = 0;
  for (i = 0; i < 10; i++) add += parseInt(cpf.charAt(i)) * (11 - i);
  rev = 11 - (add % 11);
  if (rev === 10 || rev === 11) rev = 0;
  if (rev !== parseInt(cpf.charAt(10))) return false;
  return true;
};

export const validarCNPJ = cnpj => {
  cnpj = cnpj.replace(/[^\d]+/g, "");

  if (cnpj === "") return false;

  if (cnpj.length !== 14) return false;

  // Elimina CNPJs invalidos conhecidos
  if (
    cnpj === "00000000000000" ||
    cnpj === "11111111111111" ||
    cnpj === "22222222222222" ||
    cnpj === "33333333333333" ||
    cnpj === "44444444444444" ||
    cnpj === "55555555555555" ||
    cnpj === "66666666666666" ||
    cnpj === "77777777777777" ||
    cnpj === "88888888888888" ||
    cnpj === "99999999999999"
  )
    return false;

  // Valida DVs
  let i;
  let tamanho = cnpj.length - 2;
  let numeros = cnpj.substring(0, tamanho);
  let digitos = cnpj.substring(tamanho);
  let soma = 0;
  let pos = tamanho - 7;
  for (i = tamanho; i >= 1; i--) {
    soma += numeros.charAt(tamanho - i) * pos--;
    if (pos < 2) pos = 9;
  }
  let resultado = soma % 11 < 2 ? 0 : 11 - (soma % 11);
  if (resultado !== parseInt(digitos.charAt(0))) return false;

  tamanho = tamanho + 1;
  numeros = cnpj.substring(0, tamanho);
  soma = 0;
  pos = tamanho - 7;
  for (i = tamanho; i >= 1; i--) {
    soma += numeros.charAt(tamanho - i) * pos--;
    if (pos < 2) pos = 9;
  }
  resultado = soma % 11 < 2 ? 0 : 11 - (soma % 11);
  if (resultado !== parseInt(digitos.charAt(1))) return false;

  return true;
};

export const hasNumber = myString => {
  return /\d/.test(myString);
};
