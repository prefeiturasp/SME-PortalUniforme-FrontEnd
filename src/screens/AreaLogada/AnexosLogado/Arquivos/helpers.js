export const verificarSeFaltamArquivos = (empresa, tiposDocumentos) => {
  let aindaFaltaDocumentoObrigatorio = false;
  tiposDocumentos.forEach((tipoDocumento) => {
    if (
      tipoDocumento.obrigatorio &&
      !empresa.arquivos_anexos.find(
        (arquivo) => arquivo.tipo_documento.id === tipoDocumento.id
      )
    )
      aindaFaltaDocumentoObrigatorio = true;
  });
  let aindaFaltamArquivos =
    empresa.lojas.find((loja) => loja.foto_fachada === null) ||
    empresa.arquivos_anexos.length === 0 ||
    aindaFaltaDocumentoObrigatorio;
  return aindaFaltamArquivos;
};
