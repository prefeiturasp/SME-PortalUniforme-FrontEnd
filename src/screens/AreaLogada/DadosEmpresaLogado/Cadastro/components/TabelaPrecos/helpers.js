export const getPrecoVezesQuantidade = (values, uniforme) => {
  if (values[uniforme.nome]) {
    const valor =
      parseFloat(values[uniforme.nome].replace(",", ".")) * uniforme.quantidade;
    return valor
      .toFixed(2)
      .toString()
      .replace(".", ",");
  } else {
    return "0,00";
  }
};

export const getTotal = (tipoDeUniforme, values) => {
  let total = 0;
  tipoDeUniforme.uniformes.forEach((uniforme) => {
    if (values[uniforme.nome]) {
      total +=
        parseFloat(values[uniforme.nome].replace(",", ".")) *
        uniforme.quantidade;
    }
  });
  return total
    .toFixed(2)
    .toString()
    .replace(".", ",");
};

export const maiorQueLimite = (tipoDeUniforme, values, limites) => {
  let total = 0;
  tipoDeUniforme.uniformes.forEach((uniforme) => {
    if (values[uniforme.nome]) {
      total +=
        parseFloat(values[uniforme.nome].replace(",", ".")) *
        uniforme.quantidade;
    }
  });
  const preco_maximo = limites.find(
    (limite) => limite.categoria_uniforme === tipoDeUniforme.id
  ).preco_maximo;
  if (total > parseFloat(preco_maximo)) {
    return preco_maximo.replace(".", ",");
  } else return false;
};
