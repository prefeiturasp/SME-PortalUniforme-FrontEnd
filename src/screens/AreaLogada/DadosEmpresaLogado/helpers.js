import { maiorQueLimite } from "./Cadastro/components/TabelaPrecos/helpers";

const addCidadeEstadoSP = (empresa) => {
  empresa.lojas.map((loja) => {
    loja.cidade = "São Paulo";
    loja.uf = "SP";
    return loja;
  });
  return empresa;
};

const formataPrecos = (empresa) => {
  let categorias = [];
  empresa.ofertas_de_uniformes.forEach((oferta) => {
    if (!categorias.includes(oferta.uniforme_categoria)) {
      categorias.push(oferta.uniforme_categoria);
    }
  });
  categorias.forEach((categoria) => {
    empresa[categoria] = true;
  });
  empresa.ofertas_de_uniformes.forEach((oferta) => {
    empresa[oferta.nome] = oferta.preco.replace(".", ",");
  });
  return empresa;
};

export const formataEmpresa = (empresa) => {
  empresa = addCidadeEstadoSP(empresa);
  empresa = formataPrecos(empresa);
  return empresa;
};

export const formataPayloadLojasPrecos = (values, tiposDeUniforme) => {
  let ofertas_de_uniformes = [];
  tiposDeUniforme.forEach((tipoDeUniforme) => {
    tipoDeUniforme.uniformes.forEach((uniforme) => {
      if (values[uniforme.nome]) {
        ofertas_de_uniformes.push({
          nome: uniforme.nome,
          valor: parseFloat(values[uniforme.nome].replace(",", ".")),
        });
      }
    });
  });
  values.ofertas_de_uniformes = ofertas_de_uniformes;
  return values;
};

export const validaTabelaPrecos = (values, tiposDeUniforme, limites) => {
  let erro = false;
  if (
    tiposDeUniforme.filter((tipoDeUniforme) => values[tipoDeUniforme.id])
      .length === 0
  ) {
    erro = "É necessário fornecer ao menos um grupo de uniformes escolares";
    return erro;
  }
  tiposDeUniforme.filter(tipo => tipo.uniformes.length > 0).forEach((tipoDeUniforme) => {
    if (
      limites.find((value) => value.categoria_uniforme === tipoDeUniforme.id)
        .obrigatorio &&
      maiorQueLimite(tipoDeUniforme, values, limites)
    ) {
      erro = `O valor máximo para os itens em ${
        tipoDeUniforme.nome
      } é R$ ${maiorQueLimite(tipoDeUniforme, values, limites)}`;
      return erro;
    }
  });
  return erro;
};
