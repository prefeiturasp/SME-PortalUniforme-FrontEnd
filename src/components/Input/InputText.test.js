import { mount } from "enzyme";
import React from "react";

import { InputText } from "./InputText";
import { required } from "helpers/fieldValidators";

describe("test <InputText>", () => {
  let wrapper;
  let wrappercustomChange;
  let onChange = jest.fn();
  let input = { onChange, value: "" };
  beforeAll(() => {
    const props = {
      meta: {
        touched: true,
        error: "This field is required"
      }
    };
    wrappercustomChange = mount(
      <InputText
        customChange={e => {}}
        className="teste"
        name="myCombo"
        input={input}
        label="mylabel"
        validate={required}
        required
        {...props}
      />
    );
    wrapper = mount(
      <InputText
        className="teste"
        name="myCombo"
        input={input}
        label="mylabel"
        validate={required}
        required
        {...props}
      />
    );
  });

  it("renders component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders component customChange", () => {
    expect(wrappercustomChange).toMatchSnapshot();
  });

  it("Change Value", () => {
    wrapper
      .find("input")
      .simulate("change", { target: { value: "Your new Value" } });
  });
});
