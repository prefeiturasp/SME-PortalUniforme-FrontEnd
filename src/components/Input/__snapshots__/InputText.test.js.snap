// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <InputText> renders component 1`] = `
<InputText
  className="teste"
  disabled={false}
  esconderAsterisco={false}
  helpText=""
  input={
    Object {
      "onChange": [MockFunction],
      "value": "",
    }
  }
  label="mylabel"
  labelClassName=""
  meta={
    Object {
      "error": "This field is required",
      "touched": true,
    }
  }
  name="myCombo"
  placeholder=""
  required={true}
  type="text"
  validate={[Function]}
>
  <div
    className="input"
  >
    <span
      className="required-asterisk"
      key="1"
    >
      *
    </span>
    <label
      className="col-form-label "
      htmlFor="myCombo"
      key="2"
    >
      mylabel
    </label>
    <input
      className="form-control teste invalid-field"
      disabled={false}
      name="myCombo"
      onChange={[MockFunction]}
      placeholder=""
      required={true}
      type="text"
      value=""
    />
    <HelpText
      helpText=""
    >
      <div
        className="help-text"
      />
    </HelpText>
    <InputErroMensagem
      meta={
        Object {
          "error": "This field is required",
          "touched": true,
        }
      }
    >
      <div
        className="error-or-warning-message"
      >
        <div
          className="error-message"
        >
          This field is required
        </div>
      </div>
    </InputErroMensagem>
  </div>
</InputText>
`;

exports[`test <InputText> renders component customChange 1`] = `
<InputText
  className="teste"
  customChange={[Function]}
  disabled={false}
  esconderAsterisco={false}
  helpText=""
  input={
    Object {
      "onChange": [MockFunction],
      "value": "",
    }
  }
  label="mylabel"
  labelClassName=""
  meta={
    Object {
      "error": "This field is required",
      "touched": true,
    }
  }
  name="myCombo"
  placeholder=""
  required={true}
  type="text"
  validate={[Function]}
>
  <div
    className="input"
  >
    <span
      className="required-asterisk"
      key="1"
    >
      *
    </span>
    <label
      className="col-form-label "
      htmlFor="myCombo"
      key="2"
    >
      mylabel
    </label>
    <input
      className="form-control teste invalid-field"
      disabled={false}
      name="myCombo"
      onChange={[Function]}
      placeholder=""
      required={true}
      type="text"
      value=""
    />
    <HelpText
      helpText=""
    >
      <div
        className="help-text"
      />
    </HelpText>
    <InputErroMensagem
      meta={
        Object {
          "error": "This field is required",
          "touched": true,
        }
      }
    >
      <div
        className="error-or-warning-message"
      >
        <div
          className="error-message"
        >
          This field is required
        </div>
      </div>
    </InputErroMensagem>
  </div>
</InputText>
`;
